File: R/lstm-output-gate.R
sigmoid <- function(x) {
  1 / (1 + exp(-pmin(pmax(x, -500), 500)))
}

output_gate <- function(h_prev, x_t, C_t, W_o, b_o) {
  concat <- cbind(h_prev, x_t)
  o_t <- sigmoid(concat %*% t(W_o) + b_o)
  h_t <- o_t * tanh(C_t)
  list(o_t = o_t, h_t = h_t)
}

File: Julia/lstm-output-gate.jl
sigmoid(x) = 1 ./ (1 .+ exp.(-clamp.(x, -500, 500)))

function output_gate(h_prev, x_t, C_t, W_o, b_o)
    concat = hcat(h_prev, x_t)
    o_t = sigmoid(concat * W_o' .+ b_o)
    h_t = o_t .* tanh.(C_t)
    return (o_t = o_t, h_t = h_t)
end
